newrelic-agent/src/main/java/com/newrelic/agent/errors/ErrorServiceImpl.java

```diff
@@ -438,7 +456,9 @@ private void noticeTransaction(TransactionData td, TransactionStats transactionS
     private TracedError createTracedError(final String theAppName, TransactionData td, Throwable throwable, int responseStatus, String statusMessage) {
         TracedError error;
         // noticeError(expected = true)?
-        boolean markedExpected = td.getThrowable() == null ? false : td.getThrowable().expected;
+        boolean responseStatusExpected = errorCollectorConfig.getExpectedStatusCodes().contains(responseStatus);
```

Comment:
In ErrorAnalyzerImpl#isExpectedError we also do some verification on both the exception and status code.
But if the exception is not expected, it assumes the error is not expected.
Maybe we have that other check for a different report, but I believe it should follow the same rules.

Reply:
That's interesting...We seem to return early here if the status code is expected:
https://github.com/newrelic/newrelic-java-agent/blob/e25fa98a0967879e0da0f0deaec2ade3ca7fa167/newrelic-agent/src/main/java/com/newrelic/agent/errors/ErrorAnalyzerImpl.java#L102

But, the confusing part is here where we loop through `throwable != null` and do some additional checking, but only on the throwable and not for expected status code.
https://github.com/newrelic/newrelic-java-agent/blob/e25fa98a0967879e0da0f0deaec2ade3ca7fa167/newrelic-agent/src/main/java/com/newrelic/agent/errors/ErrorAnalyzerImpl.java#L107
There is only one TransactionThrowable, so I don't fully understand these lines.  I will step through it and see.

Reply:
@meiao After stepping through this, I believe this line here:
https://github.com/newrelic/newrelic-java-agent/blob/e25fa98a0967879e0da0f0deaec2ade3ca7fa167/newrelic-agent/src/main/java/com/newrelic/agent/errors/ErrorAnalyzerImpl.java#L102

Is enough to always confirm that the status has been checked in the ErrorAnalyzer.  The status code  parameter for the method is the same as what would be on the Throwable (from what I'm seeing).  Unless, you see some way of breaking it that I'm not understanding.

Reply:
Nevermind, the fail fast threw me off. I read it the previous block as if the exception is not expected the error is not expected.

Memory must have played some trick on me, because I thought there was some two step verification in someplace from when I first investigated this issue.